### PR TITLE
Implemented clone() method in chunk and ellipsoidal_chunk geometries

### PR DESCRIPTION
**Pull request by lkaratun**
Implemented clone() method in chunk and ellipsoidal_chunk geometries (they are now pure virtual in the parent class)

### CODE REVIEW

File: source/geometry_model/chunk.cc [hunk 1]
```diff
@@ -167,6 +167,14 @@ namespace aspect
       point1_lon = p1_lon;
     }
 
+    template <int dim>
+    std::unique_ptr<Manifold<dim,dim>>
+    Chunk<dim>::ChunkGeometry::
+    clone() const
+    {
+        return std_cxx14::make_unique<ChunkGeometry>();
```
Comment (bangerth) [inline source/geometry_model/chunk.cc]: That satisfies the compiler, but it's not a copy of the current object. You'll have to somehow set the contents of the chunk geometry object to be a copy of the current object.

File: source/geometry_model/ellipsoidal_chunk.cc [hunk 2]
```diff
@@ -184,6 +184,13 @@ namespace aspect
       return push_forward_ellipsoid (push_forward_topography(chart_point), semi_major_axis_a, eccentricity);
     }
 
+    template <int dim>
+    std::unique_ptr<Manifold<dim,3>>
+    EllipsoidalChunk<dim>::EllipsoidalChunkGeometry::clone() const
+    {
+        return std_cxx14::make_unique<EllipsoidalChunkGeometry>();
```
Comment (bangerth) [inline source/geometry_model/ellipsoidal_chunk.cc]: same here

**Comment by lkaratun** [comment]
I did another attempt, please have a look @bangerth

File: source/geometry_model/chunk.cc [hunk 3]
```diff
@@ -41,6 +41,13 @@ namespace aspect
       point1_lon(0.0)
     {}
 
+    template <int dim>
+    Chunk<dim>::ChunkGeometry::ChunkGeometry(const ChunkGeometry &other, const double para_point1_lon)
+      :
+      ChartManifold<dim,dim>(other),
+      point1_lon(para_point1_lon)
```
Comment (bangerth) [inline source/geometry_model/chunk.cc]: Couldn't you omit the second argument to the constructor and simply write
```
template <int dim>
￼ +    Chunk<dim>::ChunkGeometry::ChunkGeometry(const ChunkGeometry &other, const double para_point1_lon)
￼ +      :
￼ +      ChartManifold<dim,dim>(other),
￼ +      point1_lon(other.para_point1_lon)
```
? This way, you don't have to pass all of those arguments by hand. Same for the other class that has even more of these arguments.

Comment (lkaratun) [reply source/geometry_model/chunk.cc]: Yes, I realized it right after I pushed the changes =( Fixed now

**Comment by MFraters** [comment]
Looks good to me! Please run `make indent` and squash it into logical commit blocks.

**Comment by lkaratun** [comment]
I actually ran `make indent` before pushing the changes. Got a warning regarding astyle version, but I thought it still worked. Can I not use astyle 3.1 with Aspect?

**Comment by MFraters** [comment]
Unfortunately the version is important with astyle, you will need version 2.04

**Comment by lkaratun** [comment]
astyle 2.04 doesn't seem to work: I'm getting a lot of `_: dos_to_unix: command not found` errors

**Comment by bangerth** [comment]
If you follow the link to the failed astyle test (here: http://www.math.clemson.edu/~heister/cib/aspect-astyle-hd/05d5986d4dacd13019a0e260740c1191dac07971/), then you see a diff file that shows the differences between your indentation and the one we want.You can apply that by hand as well.

(I'll note that astyle 2.04 didn't know C++11, and it gets quite confused by `std::unique_ptr<Something<dim>>`. If you want a nicer indentation, write this in the pre-C++11 style: `std::unique_ptr<Something<dim> >` -- note the space between `> >`.)

**Comment by lkaratun** [comment]
Does the checker want the lines marked with minuses or pluses? `a` and `b` aren't very descriptive names to figure out which is which =/
If it's the minuses, then I'm not sure what it's unhappy about -- both inspecting the file in the `Files changed` tab and in my text editor shows that all lines are equally indented.

P.s. astyle didn't work even after I inserted the spaces in the `std::unique_ptr`

**Comment by lkaratun** [comment]
@gassmoeller it quite confusing because after the previous failed check I thought it wanted the lines marked with minuses, and made the corresponding  changes, and the checker is happy about all but one of those changes now. And my editor was set to spaces, 2 per indentation level, which is I think similar to the rest of the Aspect code. 
Anyway, I ran the `patch `command as you suggested, it's much easier than editing by hand, thank you.

**Comment by gassmoeller** [comment]
Hmm, not sure what happened then. Do you by chance make the modifications on a windows system? That could introduce problems with the line endings, but apart from that I am not sure what could cause the issues. Anyway, the patch should fix it.

**Comment by gassmoeller** [comment]
Thanks for the fix!

**Pull request merged**